src/entrypoints/entrypoints.ts

```diff
@@ -0,0 +1,144 @@
+import { Address } from "../address";
+import { AccountController } from "../controllers/accountController";
+import { DelegationController } from "../controllers/delegationController";
+import { IAccount } from "../controllers/interfaces";
+import { RelayedController } from "../controllers/relayedController";
+import { SmartContractController } from "../controllers/smartContractController";
+import { TokenManagementController } from "../controllers/tokenManagementController";
+import { TransfersController } from "../controllers/transfersController";
+import { ErrInvalidNetworkProviderKind } from "../errors";
+import { Message, MessageComputer } from "../message";
+import { ApiNetworkProvider, ProxyNetworkProvider, TransactionOnNetwork } from "../networkProviders";
+import { AbiRegistry } from "../smartcontracts";
+import { Transaction } from "../transaction";
+import { TransactionComputer } from "../transactionComputer";
+import { TransactionWatcher } from "../transactionWatcher";
+import { UserVerifier } from "../wallet";
+import { DevnetEntrypointConfig, MainnetEntrypointConfig, TestnetEntrypointConfig } from "./config";
+import { ProviderWrapper } from "./providerWrapper";
+
+class NetworkEntrypoint {
+    private networkProvider: ApiNetworkProvider | ProxyNetworkProvider;
+    private chainId: string;
+
+    constructor(networkProviderUrl: string, networkProviderKind: string, chainId: string) {
```

Comment:
In TypeScript, we can have `networkProviderKind` as an enum or have the type annotation as `"proxy" | "api"` (can also stay as it is).